nilearn/_utils/param_validation.py

```diff
@@ -63,3 +68,77 @@ def check_threshold(threshold, data, percentile_func, name='threshold'):
         raise TypeError('%s should be either a number '
                         'or a string finishing with a percent sign' % (name, ))
     return threshold
+
+
+def _get_mask_volume(mask_img):
+    """Computes the volume of a brain mask in mm^3
+
+    Parameters
+    ----------
+    mask_img : nibabel image object
+        Input image whose voxel dimensions are to be computed.
+
+    Returns
+    -------
+    vol : float
+        The computed volume.
+    """
+    vox_dims = get_header(mask_img).get_zooms()[:3]
+    return 1. * np.prod(vox_dims) * mask_img.get_data().astype(np.bool).sum()
+
+
+def _adjust_screening_percentile(screening_percentile, mask_img,
+                                 verbose=0):
+    """Adjusts the screening percentile according to the MNI mask template.
+
+    Parameters
+    ----------
+    screening_percentile : float in the interval [0, 100]
+        Percentile value for ANOVA univariate feature selection. A value of
+        100 means 'keep all features'. This percentile is is expressed
+        w.r.t the volume of a standard (MNI152) brain, and so is corrected
+        at runtime by premultiplying it with the ratio of the volume of the
+        mask of the data and volume of a standard brain.  If '100' is given,
+        all the features are used, regardless of the number of voxels.
+
+    mask_img : nibabel image object
+        Input image whose voxel dimensions are to be computed.
+
+    verbose : int, optional (default 0)
+        Verbosity level.
+
+    Retruns
+    -------
+    screening_percentile: float in the interval [0, 100]
+        Percentile value for ANOVa univariate feature selection.
+    """
+    original_screening_percentile = screening_percentile
+    # correct screening_percentile according to the volume of the data mask
+    mask_volume = _get_mask_volume(mask_img)
+    if mask_volume > MNI152_BRAIN_VOLUME:
```

Comment:
I find this threshold a bit strange. This has 50% chance to occur in real life... ?
I would test if mask_volme < 1.1 * MNI152_BRAIN_VOLUME
and Mask_volume >.5 * MNI152_BRAIN_VOLUME
Maybe I'm missing something.